src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/StoreV2/DiaryDayV2.swift

```diff
@@ -0,0 +1,46 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+struct DiaryDayV2: Equatable {
```

Comment:
There is actually no need to create a new version. Just update the old struct and add a migration.

Reply:
No we can't because otherwise the tests for the migration from V1 to V2 would not run anymore. But we want to keep the tests for migrate from V1 to V2. 
In these tests, a database V1 is created, data is written in there and then the migration is called. This would not work if we only have one struct with the logic in it for V3.

Reply:
I had a similar discussion with Artur about this and in my opinion there is no reason to handle it this way. 

The migration should contain only the changes in the database structure. With that at hand, the representation objects like `DiaryDay` doesn't need to contain previous states. Same for any kind of 'logic', i.e. how data is handled within the app.

To cut it short, the current way of handling migrations is cumbersome at best. It does not follow any patterns for data migration I know of and produces tons of redundant code by copy/paste existing. The latter also prevents us from tracking changes in the 'root' class.